examples/keyvalue-inmemory/main.go

```diff
@@ -19,22 +19,38 @@ func main() {
 }
 
 func run() error {
+	handler := Handler{}
 	p, err := provider.New(
-		provider.SourceLinkPut(handleNewSourceLink),
-		provider.TargetLinkPut(handleNewTargetLink),
-		provider.SourceLinkDel(handleDelSourceLink),
-		provider.TargetLinkDel(handleDelTargetLink),
-		provider.HealthCheck(handleHealthCheck),
-		provider.Shutdown(handleShutdown),
+		provider.SourceLinkPut(func(link provider.InterfaceLinkDefinition) error {
+			return handleNewSourceLink(&handler, link)
+		}),
+		provider.TargetLinkPut(func(link provider.InterfaceLinkDefinition) error {
+			return handleNewTargetLink(&handler, link)
+		}),
+		provider.SourceLinkDel(func(link provider.InterfaceLinkDefinition) error {
+			return handleDelSourceLink(&handler, link)
+		}),
+		provider.TargetLinkDel(func(link provider.InterfaceLinkDefinition) error {
+			return handleDelTargetLink(&handler, link)
+		}),
+		provider.HealthCheck(func() string {
+			return handleHealthCheck(&handler)
+		}),
+		provider.Shutdown(func() error {
+			return handleShutdown(&handler)
+		}),
 	)
 	if err != nil {
 		return err
 	}
+
+	// Store provider reference in handler
+	handler.WasmcloudProvider = p
 	providerCh := make(chan error, 1)
 	signalCh := make(chan os.Signal, 1)
 
 	// Handle RPC operations
-	stopFunc, err := server.Serve(p.RPCClient, &Provider{})
+	stopFunc, err := server.Serve(p.RPCClient, &Handler{})
```

Comment:
You're not passing the handler that you created above to the server. This is passing a reference to an empty `Handler`.
```
stopFunc, err := server.Serve(p.RPCClient, handler)
```